tasks/generate-sitemap.mjs

```diff
@@ -1,37 +1,53 @@
-// Adapted from https://gist.github.com/joranquinten/78f3e288274a3c9405a499b8a8c46e35
-import * as fs from 'fs';
+import { promises as fs } from 'fs';
 import { execSync } from 'child_process';
-import directory from '../src/directory/directory.json' assert { type: 'json' };
+import crypto from 'node:crypto';
+import * as cheerio from 'cheerio';
+import dotenv from 'dotenv';
+import flatDirectory from '../src/directory/flatDirectory.json' assert { type: 'json' };
+
+dotenv.config({ path: './.env.custom' });
+
+const SITEMAP_DOMAIN = process.env.SITEMAP_DOMAIN
+  ? process.env.SITEMAP_DOMAIN
+  : 'https://docs.amplify.aws';
+
+// Path of the Next.js static HTML build output
+const ROOT_PATH = './client/www/next-build';
 
 const formatDate = (date) => `${date.toISOString().split('.')[0]}+0:00`;
 const getPriority = () => 0.5;
 
-const lastModifiedCache = {};
-const lastModified = (path) => {
-  if (fs.existsSync(path)) {
-    if (path in lastModifiedCache) {
-      return lastModifiedCache[path];
-    }
-    return (lastModifiedCache[path] = execSync(
-      `git log -1 --pretty="format:%cI" ${path}`
-    ));
-  }
-  console.log(`${path} doesn't exist`);
-  return formatDate(new Date());
-};
+const PLATFORMS = [
```

Comment:
can we reuse the platform const from https://github.com/aws-amplify/docs/blob/main/src/data/platforms.ts

Reply:
When I tried to do that originally, it kept throwing an error saying it couldn't find the module `@data/platforms`. I also tried specifying the relative path `../src/data/platforms.js` but that didn't work either. Maybe I'm doing something wrong though? Do you have any thoughts?

Reply:
I wonder if it's because `platforms.ts` is a typescript file and it needs to be a complied js file? 🤔